payjoin/src/core/error.rs

```diff
@@ -1,3 +1,65 @@
-use std::error;
+use std::{error, fmt};
 
-pub type ImplementationError = Box<dyn error::Error + Send + Sync>;
+use crate::receive::OutputSubstitutionError;
+
+#[derive(Debug)]
+pub struct ImplementationError(Box<dyn error::Error + Send + Sync>);
+
+impl ImplementationError {
+    pub fn new(e: impl error::Error + Send + Sync + 'static) -> Self {
+        ImplementationError(Box::new(e))
+    }
+}
```

Comment:
While I do believe such an `ImplementationError` is appropriate, I'm doubt that all of the `From` implementations, which are all part of the public API are when the simple `Implementation::new` type can be used.